Justfile

```diff
@@ -10,6 +10,31 @@ release-binary:
     cargo build --release
     @just copy-binary
 
+# Build Windows executable
+release-windows:
+    #!/usr/bin/env sh
+    if [ "$(uname)" = "Darwin" ] || [ "$(uname)" = "Linux" ]; then
+        echo "Building Windows executable using Docker..."
+        docker volume create goose-windows-cache || true
+        docker run --rm \
+            -v "$(pwd)":/usr/src/myapp \
+            -v goose-windows-cache:/usr/local/cargo/registry \
+            -w /usr/src/myapp \
+            rust:latest \
+            sh -c "rustup target add x86_64-pc-windows-gnu && \
+                apt-get update && \
+                apt-get install -y mingw-w64 && \
+                cargo build --release --target x86_64-pc-windows-gnu && \
+                GCC_DIR=\$(ls -d /usr/lib/gcc/x86_64-w64-mingw32/*/ | head -n 1) && \
+                cp \$GCC_DIR/libstdc++-6.dll /usr/src/myapp/target/x86_64-pc-windows-gnu/release/ && \
+                cp \$GCC_DIR/libgcc_s_seh-1.dll /usr/src/myapp/target/x86_64-pc-windows-gnu/release/ && \
+                cp /usr/x86_64-w64-mingw32/lib/libwinpthread-1.dll /usr/src/myapp/target/x86_64-pc-windows-gnu/release/"
+    else
+        echo "Building Windows executable using Docker through PowerShell..."
+        powershell.exe -Command "docker volume create goose-windows-cache; docker run --rm -v ${PWD}:/usr/src/myapp -v goose-windows-cache:/usr/local/cargo/registry -w /usr/src/myapp rust:latest sh -c 'rustup target add x86_64-pc-windows-gnu && apt-get update && apt-get install -y mingw-w64 && cargo build --release --target x86_64-pc-windows-gnu && GCC_DIR=\$(ls -d /usr/lib/gcc/x86_64-w64-mingw32/*/ | head -n 1) && cp \$GCC_DIR/libstdc++-6.dll /usr/src/myapp/target/x86_64-pc-windows-gnu/release/ && cp \$GCC_DIR/libgcc_s_seh-1.dll /usr/src/myapp/target/x86_64-pc-windows-gnu/release/ && cp /usr/x86_64-w64-mingw32/lib/libwinpthread-1.dll /usr/src/myapp/target/x86_64-pc-windows-gnu/release/'"
```

Comment:
What's happening here? 😃  Please help me understand lol. Might be easier to make a script and call that

Reply:
essentially I added this to build it on docker as I could not compile windows executable on my mac. 
This does the windows specific build + copy of required dll's 
I tried extracting to a separate script, but it broke so many times I gave up for now.